src/strongbox/nfo.clj

```diff
@@ -139,6 +141,44 @@
        (s/valid? :addon/nfo (utils/load-json-file-safely (nfo-path install-dir (:dirname addon))
                                                          {:transform-map {:installed-game-track keyword}}))))
 
+(defn-spec update-nfo nil?
+  "updates *existing* nfo data with new values.
+  content is only written if valid."
+  [install-dir ::sp/extant-dir, dirname ::sp/dirname, updates map?]
+  (let [path (nfo-path install-dir dirname)
+        nfo (read-nfo install-dir dirname)
+        new-nfo (merge nfo updates)
+        ;; convenience for implicitly ignored addons.
+        ;; if the nfo file was missing and the new request is to remove the flag, set it to false instead
+        new-nfo (if (and (not (fs/exists? path))
+                         (= {:ignore? nil} new-nfo))
+                  {:ignore? false}
+                  new-nfo)
+        ]
+    (if (= {:ignore? nil} new-nfo)
+      ;; edge case. a valid nfo file is also simply a `ignore: True`
+      ;; when this is true, just delete the nfo file
+      (rm-nfo path)
+      (when (s/valid? :addon/nfo new-nfo)
```

Comment:
make this an `if` with an error case. I want to know, even if it's in debug, if updating the nfo data fails